shared/utility/math/geometry/UnitQuaternion.hpp

```diff
@@ -24,128 +24,125 @@
+            double norm();
 
-                // real part
-                inline double kW() const {
-                    return at(0);
-                };
-                inline double& kW() {
-                    return at(0);
-                };
-
-                inline double kX() const {
-                    return at(1);
-                };
-                inline double& kX() {
-                    return at(1);
-                };
-
-                inline double kY() const {
-                    return at(2);
-                };
-                inline double& kY() {
-                    return at(2);
-                };
+            // real part
+            inline double kW() const {
+                return at(0);
+            };
+            inline double& kW() {
+                return at(0);
+            };
 
-                inline double kZ() const {
-                    return at(3);
-                };
-                inline double& kZ() {
-                    return at(3);
-                };
+            inline double kX() const {
+                return at(1);
+            };
+            inline double& kX() {
+                return at(1);
+            };
 
-                inline double real() const {
-                    return at(0);
-                };
-                inline double& real() {
-                    return at(0);
-                };
+            inline double kY() const {
+                return at(2);
+            };
+            inline double& kY() {
+                return at(2);
+            };
 
-                inline const arma::subview_col<double> imaginary() const {
-                    return rows(1, 3);
-                }
-                inline arma::subview_col<double> imaginary() {
-                    return rows(1, 3);
-                }
+            inline double kZ() const {
+                return at(3);
+            };
+            inline double& kZ() {
+                return at(3);
+            };
 
-                UnitQuaternion slerp(const UnitQuaternion& p, const double& t);
-                static inline UnitQuaternion Identity() {
-                    return (arma::vec4({1.0, 0.0, 0.0, 0.0}));
-                }
+            inline double real() const {
+                return at(0);
+            };
+            inline double& real() {
+                return at(0);
             };
-        }  // namespace geometry
-    }      // namespace math
-}  // namespace utility
 
+            inline const arma::subview_col<double> imaginary() const {
+                return rows(1, 3);
+            }
+            inline arma::subview_col<double> imaginary() {
+                return rows(1, 3);
+            }
+
+            UnitQuaternion slerp(const UnitQuaternion& p, const double& t);
+            static inline UnitQuaternion Identity() {
+                return (arma::vec4({1.0, 0.0, 0.0, 0.0}));
+            }
+        };
+    }  // namespace geometry
```

Comment:
This shouldn't be here

Reply:
It actually should though
the way this was written, it has two namespaces within that one. We _could_ fully qualify each of them, but that seems less nice than the current situation.
The author did something similar in `Rotation3D`, as well as `Configuration.hpp`, `FileWatch.hpp` and `Script.hpp` :p